Frameworks/UIKit/UISearchBar.mm

```diff
@@ -72,17 +72,33 @@ - (BOOL)resignFirstResponder {
  @Status Interoperable
 */
 - (void)setPrompt:(NSString*)prompt {
-    if (_promptLabel == nil) {
-        CGRect promptFrame = CGRectMake(0, c_marginTopForPrompt, self.frame.size.width, c_defaultTextFieldHeight);
-
-        self->_promptLabel.attach([[UILabel alloc] initWithFrame:promptFrame]);
-        [self->_promptLabel setTextAlignment:UITextAlignmentCenter];
-        [self->_promptLabel setBackgroundColor:nil];
-        [self->_promptLabel setTextColor:[UIColor blackColor]];
-        [self addSubview:self->_promptLabel];
+    if (_prompt == prompt || (([_prompt length] == 0) && ([prompt length] == 0))) {
```

Comment:
>_prompt == prompt [](start = 8, length = 17)

don't we want to do a string comparison here?  in which case, wouldn't it also work for the check against an empty string?